Sources/Ink/Internal/URLDeclaration.swift

```diff
@@ -7,14 +7,29 @@
 internal struct URLDeclaration: Readable {
     var name: String
     var url: URL
+    var title: Substring?
 
     static func read(using reader: inout Reader) throws -> Self {
         try reader.read("[")
         let name = try reader.read(until: "]")
         try reader.read(":")
         try reader.readWhitespaces()
-        let url = reader.readUntilEndOfLine()
 
-        return URLDeclaration(name: name.lowercased(), url: url)
+        var titleText: Substring? = nil
+        let url = try reader.readCharacters(matching: \.isSameLineNonWhitespace)
+
+        if !reader.didReachEnd {
+			if reader.currentCharacter.isNewline {
+				reader.advanceIndex()
+			}
+			if reader.currentCharacter.isSameLineWhitespace {
+				try reader.readWhitespaces()
+			}
+            if let delimeter = TitleDelimeter(rawValue: reader.currentCharacter) {
```

Comment:
```suggestion
            if let delimeter = TitleDelimeter(rawValue: reader.currentCharacter) {
                let index = reader.currentIndex
```

We capture the index at the start of the title parsing, in case there is an otherwise valid title followed by non-whitespace. If that happens, we rewind the reader to the beginning of what would have been the title, and set `titleText = nil`.